host_source.go

```diff
@@ -271,106 +365,279 @@ func checkSystemLocal(control *controlConn) (bool, error) {
 	return true, nil
 }
 
-// Returns true if we are using system_schema.keyspaces instead of system.schema_keyspaces
-func checkSystemSchema(control *controlConn) (bool, error) {
-	iter := control.query("SELECT * FROM system_schema.keyspaces")
-	if err := iter.err; err != nil {
-		if errf, ok := err.(*errorFrame); ok {
-			if errf.code == errReadFailure {
-				return false, nil
+// Given a map that represents a row from either system.local or system.peers
+// return as much information as we can in *HostInfo
+func (r *ringDescriber) hostInfoFromMap(row map[string]interface{}) (error, *HostInfo) {
```

Comment:
change the return definition to be `(*HostInfo, error)` to be consistent with Go code style